pkg/reconciler/v1alpha1/autoscaling/hpa/hpa.go

```diff
@@ -119,11 +120,12 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		// This is important because the copy we loaded from the informer's
 		// cache may be stale and we don't want to overwrite a prior update
 		// to status with this stale state.
-	} else {
-		if _, err := c.updateStatus(pa); err != nil {
-			logger.Warnw("Failed to update pa status", zap.Error(err))
-			return err
-		}
+	} else if _, err := c.updateStatus(pa); err != nil {
+		logger.Warnw("Failed to update pa status", zap.Error(err))
+		return err
+	}
+	if err != nil {
+		c.Recorder.Eventf(pa, corev1.EventTypeWarning, "InternalError", err.Error())
```

Comment:
Here and elsewhere do we want to report the location? Or is it automatically reported?

Reply:
Code location isn't really interesting since this appears in user-facing resources.

Reply:
OK lemme rephrase as more like: which `reconciler` failed? Since they all look the same.